operator/api/auto/v1alpha1/workspace_types.go

```diff
@@ -34,11 +34,6 @@ const (
 	SecurityProfileBaseline SecurityProfile = "baseline"
 	// SecurityProfileRestricted applies the restricted security profile.
 	SecurityProfileRestricted SecurityProfile = "restricted"
-
-	// SecurityProfileBaselineDefaultImage is the default image used when the security profile is 'baseline'.
-	SecurityProfileBaselineDefaultImage = "pulumi/pulumi:latest"
-	// SecurityProfileRestrictedDefaultImage is the default image used when the security profile is 'restricted'.
-	SecurityProfileRestrictedDefaultImage = "pulumi/pulumi:latest-nonroot"
```

Comment:
Rationale: moving these constants and the associated 'defaulting' logic to `webhook/auto/v1alpha1` where, in the future, a true webhook would apply the defaults eagerly.  This PR does NOT implement a webhook, it simply applies the defaults during reconciliation for simplicity.

Reply:
Let me say, the benefit of applying defaults eagerly (with a webhook) rather than lazily (during reconciliation) is stability; one may change the default later without affecting existing workloads.